src/plugins/ciu/ciuPlugin.js

```diff
@@ -0,0 +1,73 @@
+const db = require('caniuse-db/data.json');
+const { messageToFactoid } = require('../factoids/factoidsPlugin');
+
+const featureKeys = Object.keys(db.data);
+
+const AGENTS_TO_SHOW = {
+  ie: 'IE',
+  edge: 'Edge',
+  firefox: 'FF',
+  chrome: 'Chrome',
+  opera: 'Opera',
+  safari: 'Safari',
+  ios_saf: 'iOS',
+  android: 'Android',
+};
+
+function getAgentSupportArray(feature) {
+  return Object.entries(AGENTS_TO_SHOW).flatMap(([agent, agentName]) => {
+    let supportedVersions = db.agents[agent].versions.filter(
+      (v) => v && !/[nupax]/g.test(feature.stats[agent][v]),
+    );
+
+    if (supportedVersions.length === 0) {
+      supportedVersions = db.agents[agent].versions.filter(
+        (v) => v && !/[nu]/g.test(feature.stats[agent][v]),
+      );
+    }
+
+    return supportedVersions.length > 0
+      ? [`${agentName}${supportedVersions[0]}`]
+      : [];
+  });
+}
+
+module.exports = async function ciuPlugin(msg) {
+  if (!msg.command) return;
+
+  const [cmd, ...search] = msg.command.command.split(' ');
+  if (cmd !== 'ciu' || search.length === 0) {
+    return;
+  }
+  const factoid = await messageToFactoid(msg);
+  if (factoid) {
+    return;
+  }
+
+  msg.handling();
+
+  for (const term of search.map((s) => s.toLowerCase())) {
+    // search by order of importance, so keys first, then title, then go through descriptions
+    const featKey =
+      featureKeys.find((key) => key.includes(term)) ||
+      featureKeys.find((key) => db.data[key].title.toLowerCase().includes(term)) ||
+      featureKeys.find((key) =>
+        db.data[key].description.toLowerCase().includes(term),
+      );
```

Comment:
why not use entries instead of keys?

Reply:
I just tried using entries, but I don't like it more

Reply:
avoiding having to reference the object is a huge benefit

Reply:
performance benefit you mean? I really tried the other variant, but I wasn't convinced, I also even tried reshaping the feature objects, but in the end things just get more complex, I prefer sticking to the original shape

Reply:
no, performance doesn't matter; i mean a clarity benefit.